src/expansionmap.cpp

```diff
@@ -72,18 +72,19 @@ void ExpansionMap::Expand(const xsdAnyURI_List &expansionList, xsdAnyURI_List &e
                 // clone current section/ending/rdg/lem and rename it, adding -"rend2" for the first repetition etc.
                 Object *clonedObject = currSect->Clone();
                 clonedObject->CloneReset();
+                GeneratePredictableIds(currSect, clonedObject);
                 clonedObject->SetUuid(currSect->GetUuid() + "-rend"
                     + std::to_string(GetExpansionIdsForElement(currSect->GetUuid()).size() + 1));
```

Comment:
Can't you do this in `GeneratePredictableIds` before looping through the children? Then it is all in one place

Reply:
Done.